tests/loggers/test_neptune.py

```diff
@@ -119,8 +67,115 @@ def _run_training(logger):
+    logger_open_after_fit = _run_training(NeptuneLogger(api_key='test', project='project', close_after_fit=False))
+    assert logger_open_after_fit.experiment.stop.call_count == 0
+
+
+def _assert_legacy(callback, *args, **kwargs):
+    try:
+        callback(*args, **kwargs)
+    except ValueError:
+        pass
+    else:
+        raise AssertionError("Should throw `ValueError`")
+
+
+@patch('pytorch_lightning.loggers.neptune.neptune')
+def test_log_hyperparams(neptune):
+    params = {
+        'foo': 'bar',
+        'nested_foo': {
+            'bar': 42
+        }
+    }
+    test_variants = [
+        ({}, 'parameters'),
+        ({'base_namespace': 'custom_namespace'}, 'custom_namespace/parameters'),
+    ]
+    for base_namespace, parameters_key in test_variants:
+        # given:
+        logger = NeptuneLogger(api_key='test', project='project', **base_namespace)
+        logger._run_instance = MagicMock()
+        run_attr_mock = MagicMock()
+        logger._run_instance.__getitem__.return_value = run_attr_mock
+
+        # when: log parameters
+        logger.log_hyperparams(params)
+
+        # then
+        logger._run_instance.__setitem__.assert_any_call(parameters_key, params)
+
+
+@patch('pytorch_lightning.loggers.neptune.neptune')
+def test_log_metrics(neptune):
+    metrics = {
+        'foo': 42,
+        'bar': 555,
+    }
+    test_variants = [
+        ({}, ('metrics/foo', 'metrics/bar')),
+        ({'base_namespace': 'custom_namespace'}, ('custom_namespace/metrics/foo', 'custom_namespace/metrics/bar')),
+    ]
+
+    for base_namespace, (metrics_foo_key, metrics_bar_key) in test_variants:
+        # given:
+        logger = NeptuneLogger(api_key='test', project='project', **base_namespace)
+        logger._run_instance = MagicMock()
+        run_attr_mock = MagicMock()
+        logger._run_instance.__getitem__.return_value = run_attr_mock
+
+        # when: log metrics
+        logger.log_metrics(metrics)
+
+        # then:
+        logger._run_instance.__getitem__.assert_any_call(metrics_foo_key)
+        logger._run_instance.__getitem__.assert_any_call(metrics_bar_key)
+        run_attr_mock.log.assert_has_calls([call(42), call(555)])
+
+
+@patch('pytorch_lightning.loggers.neptune.neptune')
+def test_legacy_init_kwargs(neptune):
+    legacy_neptune_init_kwargs = [
+        'project_name',
+        'offline_mode',
+        'experiment_name',
+        'experiment_id',
+        'params',
+        'properties',
+        'upload_source_files',
+        'abort_callback',
+        'logger',
+        'upload_stdout',
+        'upload_stderr',
+        'send_hardware_metrics',
+        'run_monitoring_thread',
+        'handle_uncaught_exceptions',
+        'git_info',
+        'hostname',
+        'notebook_id',
+        'notebook_path',
+    ]
+    for legacy_kwarg in legacy_neptune_init_kwargs:
+        _assert_legacy(
+            NeptuneLogger,
+            **{legacy_kwarg: None}
+        )
```

Comment:
I think you could do something like this directly:
```suggestion
        with pytest.raises(ValueError, match="your message"):
            NeptuneLogger(...)
```